src/AppBundle/Controller/CoDeputyController.php

```diff
@@ -0,0 +1,183 @@
+<?php
+
+namespace AppBundle\Controller;
+
+use AppBundle\Entity as EntityDir;
+use AppBundle\Form as FormDir;
+use AppBundle\Model\SelfRegisterData;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Template;
+use Symfony\Component\Form\FormError;
+use Symfony\Component\HttpFoundation\Request;
+
+class CoDeputyController extends AbstractController
+{
+    /**
+     * @Route("/codeputy/verification", name="codep_verification")
+     * @Template()
+     */
+    public function verificationAction(Request $request)
+    {
+        $user = $this->getUserWithData(['user', 'user-clients', 'client']);
+
+        $form = $this->createForm(new FormDir\CoDeputyVerificationType(), $user);
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted()){
+
+            // get client validation errors, if any, and add to the form
+            $client = new EntityDir\Client();
+            $client->setLastName($form['clientLastname']->getData());
+            $client->setCaseNumber($form['clientCaseNumber']->getData());
+            $errors = $this->get('validator')->validate($client, ['verify-codeputy']);
+            foreach($errors as $error) {
+                $clientProperty = $error->getPropertyPath();
+                $form->get('client'.ucfirst($clientProperty))->addError(new FormError($error->getMessage()));
+            }
+
+            if ($form->isValid()) {
+
+                $selfRegisterData = new SelfRegisterData();
+                $selfRegisterData->setFirstname($form['firstname']->getData());
+                $selfRegisterData->setLastname($form['lastname']->getData());
+                $selfRegisterData->setEmail($form['email']->getData());
+                $selfRegisterData->setPostcode($form['addressPostcode']->getData());
+                $selfRegisterData->setClientLastname($form['clientLastname']->getData());
+                $selfRegisterData->setCaseNumber($form['clientCaseNumber']->getData());
+
+                // validate against casRec
+                try {
+                    $this->getRestClient()->apiCall('post', 'selfregister/verifycodeputy', $selfRegisterData, 'array', [], false);
+                    $user->setCoDeputyClientConfirmed(true);
+                    $this->getRestClient()->put('user/' . $user->getId(), $user);
+                    return $this->redirect($this->generateUrl('homepage'));
+                } catch (\Exception $e) {
+                    $translator = $this->get('translator');
```

Comment:
wondering if it's worth moving this logic in a common place, as it's used by the self register controller too..